core/src/main/java/com/linecorp/armeria/server/file/NonExistentHttpFile.java

```diff
@@ -29,16 +31,18 @@
 
 final class NonExistentHttpFile implements HttpFile {
 
-    static final NonExistentHttpFile INSTANCE = new NonExistentHttpFile(null);
+    static final NonExistentHttpFile INSTANCE = new NonExistentHttpFile(null, false);
 
     private static final CompletableFuture<AggregatedHttpFile> AGGREGATED_FUTURE =
             UnmodifiableFuture.completedFuture(NonExistentAggregatedHttpFile.INSTANCE);
 
     @Nullable
     private final String location;
+    private final boolean isRedirect;
 
-    NonExistentHttpFile(@Nullable String location) {
+    NonExistentHttpFile(@Nullable String location, boolean isRedirect) {
```

Comment:
location is always not null if `isRedirect` is false. Let's add an assertion for the contract.
```java
assert isRedirect ? location != null : true;
```